### PR TITLE
Split ZDNS into a library and CLI tool

### PR DESCRIPTION
**Pull request by phillip-stephens**
This PR exposes a new `Resolver` object so other go code can interact directly with ZDNS. `/cli` creates 1k worker threads, each with their own `Resolver` to resolve large sets of domains efficiently.

I'll add documentation of the new structure but want to get this PR merged since it's holding anything else up.

### CODE REVIEW

**Comment by phillip-stephens** [comment]
@zmap/zdns-maintainers This is ready for review but I would like to wait until we either merge or discard https://github.com/zmap/zdns/pull/370 before merging this.

**Review by zakird — APPROVED** [review]

**Comment by zakird** [comment]
@phillip-stephens just wanted to flag that I think this is in your camp at the moment as tests are currently failing :(

**Comment by phillip-stephens** [comment]
Let me actually run a linter against this. My next step after this is merged is setting up a linter CI, and that'll avoid 2 huge PR's 😄

**Pull request merged**
